src/elements/content-explorer/Content.js

```diff
@@ -46,65 +47,79 @@ type Props = {
     rootId: string,
     tableRef: Function,
     view: View,
+    viewMode?: ViewMode,
 };
 
 const Content = ({
-    view,
-    isSmall,
-    isMedium,
-    isTouch,
-    rootId,
-    rootElement,
-    currentCollection,
-    tableRef,
-    focusedRow,
-    canDownload,
     canDelete,
+    canDownload,
+    canPreview,
     canRename,
     canShare,
-    canPreview,
+    currentCollection,
+    focusedRow,
+    isMedium,
+    isSmall,
+    isTouch,
     onItemClick,
-    onItemSelect,
     onItemDelete,
     onItemDownload,
+    onItemPreview,
     onItemRename,
+    onItemSelect,
     onItemShare,
-    onItemPreview,
     onSortChange,
-}: Props) => (
-    <div className="bce-content">
-        {view === VIEW_ERROR || view === VIEW_SELECTED ? null : (
-            <ProgressBar percent={currentCollection.percentLoaded} />
-        )}
-        {isEmpty(view, currentCollection) ? (
-            <EmptyState view={view} isLoading={currentCollection.percentLoaded !== 100} />
-        ) : (
-            <ItemList
-                view={view}
-                isSmall={isSmall}
-                isMedium={isMedium}
-                isTouch={isTouch}
-                rootId={rootId}
-                rootElement={rootElement}
-                focusedRow={focusedRow}
-                currentCollection={currentCollection}
-                tableRef={tableRef}
-                canShare={canShare}
-                canPreview={canPreview}
-                canDelete={canDelete}
-                canRename={canRename}
-                canDownload={canDownload}
-                onItemClick={onItemClick}
-                onItemSelect={onItemSelect}
-                onItemDelete={onItemDelete}
-                onItemDownload={onItemDownload}
-                onItemRename={onItemRename}
-                onItemShare={onItemShare}
-                onItemPreview={onItemPreview}
-                onSortChange={onSortChange}
-            />
-        )}
-    </div>
-);
+    rootElement,
+    rootId,
+    tableRef,
+    view,
+    viewMode = VIEW_MODE_LIST,
+}: Props) => {
+    let content;
+    switch (viewMode) {
```

Comment:
The switch statement seems a bit unnecessary, since we're only dealing with two cases, could we do an inline ternary operator instead on L119?

Reply:
That would end up being a nested ternary, which is restricted by our linter, iirc. I think this is usually a sign that a component is doing too much and should probably be broken up. If that's too big a change, though, we could also create some simple state helper variables and use logical AND (`&&`), instead:

```
const isEmpty = isEmpty(view, currentcollation);
const isListView = viewMode === VIEW_MODE_LIST;
...
{isEmpty && <EmptyState view={view} isLoading={currentCollection.percentLoaded !== 100} />}
{!isEmpty && isListView && <ItemList .../>
{!isEmpty && !isListView && <GridView .../>
```